larndsim/lightLUT.py

```diff
@@ -0,0 +1,119 @@
+"""
+Code adapted from https://github.com/PPKoller/ArgonCubeLUTSim/blob/main/module_0/lutSim.py
+"""
+
+import sys, time, math
+import argparse
+import numpy as np
+from math import log, isnan
+from . import consts    
+
+def GetLutGeometry(lut_path): # Finds the max and minimum values of the x,y and z axis and the number of voxels in each direction
+
+    f = np.load(lut_path)
+    lut_min = np.array([f['Min'][0],f['Min'][1],f['Min'][2]])
+    lut_max = np.array([f['Max'][0],f['Max'][1],f['Max'][2]])
+    lut_ndiv = np.array([f['NDivisions'][0],f['NDivisions'][1],f['NDivisions'][2]])
+
+    return np.array([lut_min,lut_max,lut_ndiv])
+
+def GetVoxel(pos,lut_geometry): # Determines which voxel is being called based on the position of the edep. Voxels are indexed 0-2911
+
+    (lut_min,lut_max,lut_ndiv) = lut_geometry
+    vox_xyz = np.floor(pos/(lut_max-lut_min)*lut_ndiv).astype(int)+lut_ndiv/2
```

Comment:
I don't quite understand here, you apply the floor function, then cast to integer and then add a float? What's a typical value for `lut_ndiv`?

Reply:
this function now gets the voxel from the global coordinate system with a single int call (see https://github.com/tairaeli/larnd-sim/blob/larnd-branch/larndsim/lightLUT.py#L33)